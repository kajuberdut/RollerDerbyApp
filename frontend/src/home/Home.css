
.Home {
    width: 43%;
    margin-top: 200px;
    margin-left: 12%; 
}

.Home-Title {
  padding: 20px;
}

.Home-Link {
    padding: 5px;
}
