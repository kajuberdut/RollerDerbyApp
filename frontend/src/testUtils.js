import React from "react";
import UserContext from "./multiUse/UserContext";

const demoUser = {
    userId: 1,
    username: "testUser",
    email: "testuser@gmail.com",
    facebookName: null,
    about: null,
    primaryNumber: null,
    level: null,
    ruleset: [],
    position: [],
    insurance: [],
    locationId: null,
    associatedLeagues: null,
    phoneNumber: null,
    firstName: null,
    lastName: null,
    additionalInfo: null,
    secondaryNumber: null
  }

const UserProvider =
    ({ children, user = demoUser }) => (
    <UserContext.Provider value={{ user }}>
      {children}
    </UserContext.Provider>
);



export { UserProvider };
