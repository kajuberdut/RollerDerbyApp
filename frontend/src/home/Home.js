import React, { useEffect } from "react";
import './Home.css'
import { Link } from "react-router-dom";
import { Card, CardBody, CardText, CardTitle, Button } from "reactstrap";

/**
 * Display home page
 */

function Home({setIsHomeVis}) {

  /** Get user from context*/

  const user = JSON.parse(localStorage.getItem('user'));

  /** When home page is mounted setIsloginVis to true - false when unmounted */

  useEffect(() => {
    setIsHomeVis(true);
  return () => {
    setIsHomeVis(false);
  };
  }, []);

  /** Render home page */

    return (
      <section className="col-md-8 Home">
        <Card>
          <CardTitle className="Home-Title">
              <h1>The block party begins when the star arrives.</h1>
          </CardTitle>
          <CardBody >
            {user && <h2>Welcome to the party, {user.username}!</h2>}
            {!user && <CardText>
              <Link className="Home-Link" to="/login"><Button>Login</Button></Link>
              <Link className="Home-Link" to="/signup"><Button>Signup</Button></Link>
            </CardText>
            }
          </CardBody>
        </Card>
      </section>
    )
}

export default Home;
