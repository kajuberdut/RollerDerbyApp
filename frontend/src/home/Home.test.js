import React from "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import Home from "./Home";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";
import AllRoutes from "../routes/Routes";





// beforeEach(() => {
//   console.log("beforeEach is running")
//   console.log("window.location.pathname = '/'", window.location.pathname = "/")
//   window.location.pathname = "/";
//   window.location.href = "/"
// });

afterEach(() => {
  cleanup()
  console.log("afterEach is running")

  // render(null); // Unmount any rendered components
  // history.push("/")
})

// afterEach(() => {
//   clearHistory();
// });

describe("Tests Home Page Renders Logged Out", () => {

    test("it should render when logged out", () => {
      const { asFragment } = render(
        <BrowserRouter>
          <UserProvider user={null}>
              <Home />
          </UserProvider>
        </BrowserRouter>,
      );
    
      expect(screen.getByText('A Roller Derby Communication Application')).toBeInTheDocument();
      expect(screen.getByText('Login')).toBeInTheDocument();
      expect(screen.getByText('Signup')).toBeInTheDocument();
    }); 
  
    it("matches snapshot when logged out", function() {
      const { asFragment } = render(
        <BrowserRouter>
          <UserProvider user={null}>
              <Home />
          </UserProvider>
        </BrowserRouter>,
      );
        expect(asFragment()).toMatchSnapshot();
    });

})


describe("Tests Home Page Renders Logged In", () => {

    test("it should render when logged in", () => {
      const { asFragment } = render(
        <BrowserRouter>
          <UserProvider>
              <Home />
          </UserProvider>
        </BrowserRouter>,
      );

      expect(screen.getByText('A Roller Derby Communication Application')).toBeInTheDocument();
      expect(screen.getByText('Welcome back, testUser!')).toBeInTheDocument();
    }); 

    it("matches snapshot when logged in", function() {
      const { asFragment } = render(
        <BrowserRouter>
          <UserProvider>
              <Home />
          </UserProvider>
        </BrowserRouter>,
      );
        expect(asFragment()).toMatchSnapshot();
    });

});

describe("Tests Home Page Navigation", () => {

    test("it navigates to the signup page when clicked", () => {
      render(
        // <BrowserRouter initialEntries={["/"]}> 
        <BrowserRouter> 
          <UserProvider user={null}>
              <AllRoutes>
                  <Home />
              </AllRoutes>
          </UserProvider>
        </BrowserRouter>
      );

      expect(screen.queryByText('A Roller Derby Communication Application')).toBeInTheDocument();
      const signupLink = screen.getByText("Signup");
      fireEvent.click(signupLink);
      expect(screen.queryByText('A Roller Derby Communication Application')).not.toBeInTheDocument();
      expect(screen.getByText("Create a profile")).toBeInTheDocument();
      expect(window.location.pathname).toBe("/signup");

    });


    // test("it should render when logged in", () => {

    //   // const { asFragment } = render(
    //   //   <MemoryRouter initialEntries={["/"]}>
    //   //     <UserProvider>
    //   //         <Home />
    //   //     </UserProvider>
    //   //   </MemoryRouter>,
    //   // );
    //   // render(<Router history={history}><App/></Router>);

    //   // const history = createMemoryHistory();

    //   expect(window.location.pathname).toBe("/");
    //   expect(screen.getByText('A Roller Derby Communication Application')).toBeInTheDocument();
    //   expect(screen.getByText('Welcome back, testUser!')).toBeInTheDocument();
    // }); 


    // ! note that the last componet being rednered is the /signup which means this is not working but the test is valid. 

    // test("it navigates to the login page when clicked", () => {
    //   const { asFragment } = render(
    //     // <BrowserRouter initialEntries={["/"]}> 
    //     <MemoryRouter initialEntries={["/"]}> 
    //       <UserProvider user={null}>
    //           <AllRoutes>
    //               <Home />
    //           </AllRoutes>
    //       </UserProvider>
    //     </MemoryRouter>
    //   );
    //   expect(window.location.pathname).toBe("/");
    //   const loginLink = screen.getByText("Login");
    //   console.log("loginLink:", loginLink)
    //   expect(screen.queryByText('A Roller Derby Communication Application')).toBeInTheDocument();
    //   fireEvent.click(loginLink);
    //   expect(screen.queryByText('A Roller Derby Communication Application')).not.toBeInTheDocument();
    //   expect(screen.getByText("Login")).toBeInTheDocument();
    //   expect(window.location.pathname).toBe("/login");
    // });

});
