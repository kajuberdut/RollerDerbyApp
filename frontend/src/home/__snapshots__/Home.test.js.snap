// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Home Page Renders Logged In matches snapshot when logged in 1`] = `
<DocumentFragment>
  <section
    class="col-md-8 Home"
  >
    <div
      class="card"
    >
      <div
        class="card-title"
      >
        <h1>
          A Roller Derby Communication Application
        </h1>
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        -
        <h2>
          Welcome back, testUser!
        </h2>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Tests Home Page Renders Logged Out matches snapshot when logged out 1`] = `
<DocumentFragment>
  <section
    class="col-md-8 Home"
  >
    <div
      class="card"
    >
      <div
        class="card-title"
      >
        <h1>
          A Roller Derby Communication Application
        </h1>
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        -
        <p
          class="card-text"
        >
          <a
            class="Home-Link"
            href="/login"
          >
            <button
              class="btn btn-secondary"
            >
              Login
            </button>
          </a>
          <a
            class="Home-Link"
            href="/signup"
          >
            <button
              class="btn btn-secondary"
            >
              Signup
            </button>
          </a>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
